### PR TITLE
Fix docker hub lag on golang versions breaking canary

### LINKED ISSUE #3306: Canary broken: follow-up to #3224
### Description

As described in #3224 docker hub has a delay before go images for latest releases show-up.
The patchy-patch in #3224 was just an ad-hoc solution to fix it for rc delays, but obviously this is breaking again.

go 1.23.0 is just out, but not yet available on Hub.

**Bottom-line: today, and until Hub has the latest go image, let's ignore "canary" failures.**
@AkihiroSuda ^

I will come up with a better solution shortly.

### Steps to reproduce the issue

na

### Describe the results you received and expected

na

### What version of nerdctl are you using?

na

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

None

### Host information

_No response_

### PR DESCRIPTION
**Pull request by apostasie**
Fix #3306

### CODE REVIEW

File: hack/build-integration-canary.sh [hunk 1]
```diff
@@ -293,40 +293,55 @@ canary::build::integration(){
...
+canary::golang::hublatest(){
+  local hub_tags
+  local go_version
+  local available_version=""
+  local index
+
+  hub_tags="$(http::get /dev/stdout "https://registry-1.docker.io/v2/library/golang/tags/list" -H "Authorization: Bearer $(http::get /dev/stdout "https://auth.docker.io/token?service=registry.docker.io&scope=repository%3Alibrary%2Fgolang%3Apull" | jq  -rc .access_token)")"
+
+  index=0
+  while [ "$available_version" == "" ] && [ "$index" -lt 5 ]; do
```
Comment (ktock) [inline hack/build-integration-canary.sh:315]: nit: why `5`? Don't we need to handle cases of `available_version` isn't found?

Comment (apostasie) [reply hack/build-integration-canary.sh:315]: Limiting the number of tries to 5.
If docker hub does not have any of the past five versions, all bets are off.
Should hard error in that case though.
Will do.

Comment (apostasie) [reply hack/build-integration-canary.sh:315]: Taking that back re-reading the code. It is fine to fail to find a version. In that case, the dockerfile will just fallback to the hard-coded version.

Comment (apostasie) [reply hack/build-integration-canary.sh:318]: Latest push has the tweak.
Thanks @ktock 
Appreciate the sharp review.

Comment (apostasie) [reply hack/build-integration-canary.sh:318]: And: hard error is not necessary (hub is notorious for random failures). The new check will allow the dockerfile to still build with the hardcoded version instead.

**Review by ktock — APPROVED** [review]

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Comment by AkihiroSuda** [comment]
@ktock You may merge PRs

**Pull request merged**